### PR TITLE
Add dump_db() and dumping tokens feature to add_to_db()

### PR DESCRIPTION
**Pull request by djfedos**
features and tests are added, but there is a problem, see comment below

### CODE REVIEW

File: searchup/blocks/djfedos_db/lib_search_sdk.py [hunk 1]
```diff
@@ -14,6 +16,15 @@ def add_to_db(mdb:dict, token:str):
     if not None in cur:
         cur[None] = None
         token_added = True
+    # there is a problem with dump from here, if token is already dumped before, there will be dupes
+    # dump of whole db on each addition of a token is time-greedy to extreme
+    # and dupe check as well
+    # please advice
```
Comment (djfedos) [inline searchup/blocks/djfedos_db/lib_search_sdk.py:22]: Please note that there is a problem of dupes and bloating the dump file
At the moment I have no idea how of beautiful solution. Maybe remove this feature and just schedule a whole database dump once in a while, like once in 30 minutes or so?

**Comment by ydmeta** [comment]
Hi @djfedos you added implemtation to core,
but we need to expose those capabilities to the api
so we can call dump_db() from api,

please also add api tests in addition to core functioanlity tests

**Comment by ydmeta** [comment]
@lioryah please review this one
fix it togeter and move to #8

File: searchup/blocks/djfedos_db/lib_search_sdk.py [hunk 2]
```diff
@@ -16,15 +16,17 @@ def add_to_db(mdb:dict, token:str, dump_path:str='mdb_dump'):
     if not None in cur:
         cur[None] = None
         token_added = True
-    # there is a problem with dump from here, if token is already dumped before, there will be dupes
-    # dump of whole db on each addition of a token is time-greedy to extreme
-    # and dupe check as well
-    # please advice
+    """
```
Comment (djfedos) [inline searchup/blocks/djfedos_db/lib_search_sdk.py:19]: Note that I've disabled this dumping tokens on addition

File: searchup/blocks/djfedos_db/fapi_server.py [hunk 3]
```diff
@@ -52,10 +56,20 @@ def read_root():
 
 @app.get("/load_db/{path}")
 def load_db(path: str, q: Optional[str] = None):
+    # here is a workaround to load db from folders
```
Comment (djfedos) [inline searchup/blocks/djfedos_db/fapi_server.py:59]: I've added a workaround to load dbs from the folders over API. For me it's not convenient to put all token files to the root folder.

Comment (djfedos) [inline searchup/blocks/djfedos_db/fapi_server.py:59]: I've added a workaround to load dbs from folders to the API
It doesn't break compatibility, but adds a convenient feature

**Review by VitaliyKrivoshchapov — APPROVED** [review]

**Pull request merged**